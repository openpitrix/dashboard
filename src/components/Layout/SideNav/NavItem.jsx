import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';

import { Icon } from 'components/Base';

import styles from './index.scss';

const NavItem = ({
  className,
  to,
  iconProps,
  iconLinkCls,
  label,
  children,
  wrapLabelInLink,
  wrapIconInLink,
  ...rest
}) => {
  const labelElem = <label className={styles.title}>{label}</label>;
  const childElem = children || <Icon className={styles.icon} {...iconProps} />;

  return (
    <li className={classnames(styles.navItem, className)} {...rest}>
      {wrapIconInLink ? (
        <NavLink to={to} className={iconLinkCls}>
          {childElem}
        </NavLink>
      ) : (
        childElem
      )}

      {wrapLabelInLink ? (
        <NavLink to={to} exact>
          {labelElem}
        </NavLink>
      ) : (
        labelElem
      )}
    </li>
  );
};

NavItem.propTypes = {
  iconProps: PropTypes.object,
  to: PropTypes.string,
  wrapIconInLink: PropTypes.bool,
  wrapLabelInLink: PropTypes.bool
};

NavItem.defaultProps = {
  iconProps: {
    size: 20,
    type: 'dark'
  },
  to: '',
  wrapIconInLink: true
};

export default NavItem;
