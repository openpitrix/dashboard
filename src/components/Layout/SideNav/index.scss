@import '~scss/vars';

%disabledLink {
  color: $N75;
  opacity: 0.5;
  cursor: not-allowed;
}

.nav {
  position: fixed;
  left: 0;
  top: 0;
  z-index: 4;
  width: $menu-nav-width;
  height: 100vh;
  background-color: $N0;
  border-right: 1px solid $N10;
  transition: all 0.3s ease-out 0s;

  &:hover {
    width: $menu-width;;

    .bottomNav {
      width: $menu-width;
    }
  }

  li {
    height: 40px;
    overflow: hidden;

    a {
      color: $N300;
    }

    .icon {
      float: left;
      margin: 10px 22px;
    }

    .title {
      margin-left: -4px;
      line-height: 40px;
      cursor: pointer;
    }

    &.disabled {
      display: none;
      @extend %disabledLink;
      .title {
        cursor: not-allowed;
      }
    }

    &:hover {
      a {
        color: $P75;
      }

      .icon svg {
        --primary-color: #{$N300};
        --secondary-color: #{$P30};
      }

      :global {
        .icon-more svg {
          --primary-color: #{$P30};
        }
      }
    }
  }
}

.topNav {
  >li {
    .imageOuter {
      float: left;
      display: inline-block;
      margin: 10px 16px;
      width: 32px;
      height: 32px;
      line-height: 32px;
      opacity: 0.5;

      &.activeApp {
        opacity: 1;
        .image {
          border-color: $P75;
        }
      }
    }

    &:hover {
      .imageOuter {
        opacity: 1;
        .image {
          border-color: $N30;
        }
      }
    }

    :global {
      .icon-more {
        float: left;
        margin: 10px 22px;
        transform: rotate(90deg);
      }
    }

    &.devItem {
      height: 52px;

      .icon {
        margin: 10px 16px;
      }

      .title {
        line-height: 52px;
      }

      .addOuter {
        float: left;
        display: inline-block;
        box-sizing: border-box;
        margin: 10px 16px;
        width: 32px;
        height: 32px;

        border-radius: 2px;
        background-color: $N10;

        .icon {
          margin: 6px;
        }
      }
    }

    &:nth-child(1) {
      margin-bottom: 14px;
      height: 63px;
      border-bottom: 1px solid $N10;

      .icon {
        margin: 22px 24px 16px 24px;
      }

      .title {
        line-height: 63px;
        font-size: 16px;
        color: $N500;
        font-weight: 500;
        cursor: default;
      }
    }
  }
}


.bottomNav {
  >li {
    .iconOuter {
      [class*='target'] {
        width: auto;
        height: auto;
        border-radius: 0;
        border: 0 none;
      }

      :global{
        .pi-popover-popper{
          margin-left: -10px !important;
          margin-bottom: -30px !important;
        }
      }
    }

    .icon svg {
      opacity: 0.6;
    }

    &:last-child {
      margin-bottom: 14px;
    }
  }
}

@media screen and (min-height: 280px) {
  .bottomNav {
    position: fixed;
    bottom: 0;
    left: 0;
    z-index: 5;
    width: $menu-nav-width;
  }
}


.subNav {
  position: fixed;
  left: $menu-nav-width;
  top: 0;
  z-index: 3;
  height: 100vh;
  width: $menu-sub-nav-width;
  background-color: $N0;
  box-shadow: 0 1px 7px 0 rgba(71, 78, 93, 0.1);

  .title {
    margin: 24px 0;
    padding-left: 20px;

    .name {
      @include textCut;
      max-width: 165px;
      line-height: 16px;
      font-size: 16px;
      font-weight: 500;
      color: $N500;
    }

    .status {
      display: block;
      margin-top: 8px;
      font-size: 12px;
      color: $N75;
    }
  }

  .subContent {
    margin-bottom: 28px;
  }

  .subTitle {
    padding-left:20px;
    line-height: 24px;
    font-size: 12px;
    color: $N65;
  }

  .link {
    display: block;
    padding: 0 20px;
    line-height: 32px;
    font-size: 14px;
    color: $N300;

    &:hover {
      color: $P75;
    }

    &.active {
      font-weight: 500;
      color: $P75;
    }

    &.disabled {
      display: none;
      @extend %disabledLink;
    }
  }
}

.header{
  position: fixed;
  left: 0;
  top: 0;
  z-index: 4;
  width: 100%;
 /* position: static;
  margin: -68px -20px 20px;*/
  height: 48px;
  background-color: $background-color;

  .user [class*='target']{
    float: right;
    margin: 12px 24px;
    height: 20px;
    width: auto;
    line-height: 20px;
    color: $N500;
    font-weight: 500;
    border: 0 none;
    &:hover{
      border: 0 none;
      box-shadow: none;
      opacity: 1;
    }
    :global{
      .icon-caret-down{
        position: relative;
        top: 3px;
        left: 2px;
        svg {
          --primary-color: #{$N500};
          --secondary-color: #{$N500};
        }
      }
    }
  }

  :global{
    .pi-popover-popper{
      margin-left: -85px !important;
      margin-top: 6px !important;
    }
  }

  .operateItems {
    min-width: 120px;
    li {
      > a,
      > label{
        display: block;
        height: 32px;
        padding: 0 12px;
        //width: 100%;
        line-height: 32px;
        font-size: $font14;
        font-weight: normal;
        color: $N300;
        border-radius: 0;
        cursor: pointer;
      }

      &:hover {
        background-color: $N10;
        color: $N500;
      }

      &.line{
        border-bottom: 1px solid $N10;
      }
    }
  }
}

.headerShadow {
  box-shadow: 0 1px 4px 0 rgba(73, 33, 173, 0.06), 0 4px 8px 0 rgba(35, 35, 36, 0.04)
}


