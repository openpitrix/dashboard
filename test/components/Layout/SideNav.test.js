import { Provider } from 'mobx-react';
import { BrowserRouter } from 'react-router-dom';
import { observable } from 'mobx';

import SideNav from 'components/Layout/SideNav';

const renderPage = (store, props) => (
  <Provider rootStore={store}>
    <BrowserRouter>
      <SideNav {...props}>
        <div>test</div>
      </SideNav>
    </BrowserRouter>
  </Provider>
);

describe('Layout/SideNav', () => {
  let store,
    props;

  beforeAll(() => {
    store = observable.object({
      appStore: {
        menuApps: [
          {
            app_id: 'app_id',
            name: 'name',
            icon: 'icon'
          }
        ]
      },
      user: {
        portal: 'admin'
      },
      match: {
        appId: '123',
        path: 'app'
      }
    });

    props = {
      className: 'test',
      hasSubNav: true
    };
  });

  it('basic render', () => {
    const wrapper = render(renderPage(store, props));

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('admin portal nav', () => {
    store.user = {
      portal: 'global_admin',
      isAdmin: true,
      isAdminPortal: true
    };
    const wrapper = render(renderPage(store, props));

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('basic render ISV portal', () => {
    store.user = {
      portal: 'isv',
      isISV: true,
      isISVPortal: true
    };
    const wrapper = render(renderPage(store, props));

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('developer portal nav', () => {
    store.user = {
      portal: 'dev',
      isDev: true,
      isDevPortal: true
    };
    const wrapper = render(renderPage(store, props));

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
