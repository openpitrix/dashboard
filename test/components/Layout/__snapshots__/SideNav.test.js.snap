// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout/SideNav admin portal nav 1`] = `
<div
  class="nav"
>
  <ul
    class="topNav"
  >
    <li
      class="navItem firstElem"
    >
      <a
        href="/user/apps"
      >
        <img
          class="icon"
          src="/logo_icon.svg"
        />
      </a>
      <a
        href="/user/apps"
      >
        <label
          class="title"
        >
          Manage Console
        </label>
      </a>
    </li>
    <li
      class="navItem"
    >
      <a
        href="/admin/apps"
      >
        <span
          class="icon icon-components icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-components qicon-dark"
          >
            <use
              href="#qui-components"
            />
          </svg>
        </span>
      </a>
      <a
        href="/admin/apps"
      >
        <label
          class="title"
        >
          App Store
        </label>
      </a>
    </li>
    <li
      class="navItem"
    >
      <a
        href="/admin/providers"
      >
        <span
          class="icon icon-shield icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-shield qicon-dark"
          >
            <use
              href="#qui-shield"
            />
          </svg>
        </span>
      </a>
      <a
        href="/admin/providers"
      >
        <label
          class="title"
        >
          App service provider
        </label>
      </a>
    </li>
    <li
      class="navItem disabled"
    >
      <a
        href="/"
      >
        <span
          class="icon icon-ticket icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-ticket qicon-dark"
          >
            <use
              href="#qui-ticket"
            />
          </svg>
        </span>
      </a>
      <a
        href="/"
      >
        <label
          class="title"
        >
          Work list
        </label>
      </a>
    </li>
    <li
      class="navItem disabled"
    >
      <a
        href="/"
      >
        <span
          class="icon icon-wallet icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-wallet qicon-dark"
          >
            <use
              href="#qui-wallet"
            />
          </svg>
        </span>
      </a>
      <a
        href="/"
      >
        <label
          class="title"
        >
          Financial Center
        </label>
      </a>
    </li>
    <li
      class="navItem disabled"
    >
      <a
        href="/"
      >
        <span
          class="icon icon-linechart icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-linechart qicon-dark"
          >
            <use
              href="#qui-linechart"
            />
          </svg>
        </span>
      </a>
      <a
        href="/"
      >
        <label
          class="title"
        >
          Message and monitoring
        </label>
      </a>
    </li>
  </ul>
  <ul
    class="bottomNav"
  >
    <li>
      <div
        class="tooltip popover iconOuter"
      >
        <div
          class="target"
        >
          <span
            class="icon icon-human icon"
            style="width:20px;height:20px"
          >
            <svg
              class="qicon qicon-human qicon-dark"
            >
              <use
                href="#qui-human"
              />
            </svg>
          </span>
          <label
            class="title"
          >
            My account
          </label>
        </div>
        <div
          class="popper pi-popover-popper"
          style="position:absolute;pointer-events:none;opacity:0"
        >
          <div>
            <div
              class="pi-popover-content"
            >
              <ul
                class="menuLayer"
              >
                <li>
                  <span
                    class="userIcon"
                  >
                    <span
                      class="icon icon-human iconImg"
                      style="width:20px;height:20px"
                    >
                      <svg
                        class="qicon qicon-human qicon-dark"
                      >
                        <use
                          href="#qui-human"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="username"
                  />
                  <span
                    class="devIconOuter undefined"
                  >
                    <span
                      class="icon icon-enterprise devIcon"
                      style="width:8px;height:8px"
                    >
                      <svg
                        class="qicon qicon-enterprise qicon-white"
                      >
                        <use
                          href="#qui-enterprise"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
                <li
                  class="dev"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-appcenter icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-appcenter qicon-dark"
                      >
                        <use
                          href="#qui-appcenter"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      App Center
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/account"
                  >
                    <span
                      class="icon icon-folder icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-folder qicon-dark"
                      >
                        <use
                          href="#qui-folder"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Account Info
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/password"
                  >
                    <span
                      class="icon icon-lock icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-lock qicon-dark"
                      >
                        <use
                          href="#qui-lock"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Change Password
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-loudspeaker icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-loudspeaker qicon-dark"
                      >
                        <use
                          href="#qui-loudspeaker"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Notice settings
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-creditcard icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-creditcard qicon-dark"
                      >
                        <use
                          href="#qui-creditcard"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Payment
                    </span>
                  </a>
                </li>
                <li
                  class="divider"
                >
                  <a
                    href="/user/runtimes"
                  >
                    <span
                      class="icon icon-image icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-image qicon-dark"
                      >
                        <use
                          href="#qui-image"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Testing env
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/clusters"
                  >
                    <span
                      class="icon icon-cluster icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-cluster qicon-dark"
                      >
                        <use
                          href="#qui-cluster"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Test instance
                    </span>
                  </a>
                </li>
                <li
                  class="divider"
                >
                  <a
                    href="/logout"
                  >
                    <span
                      class="icon icon-previous icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-previous qicon-dark"
                      >
                        <use
                          href="#qui-previous"
                        />
                      </svg>
                    </span>
                    <span>
                      Log out
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Layout/SideNav basic render 1`] = `
<div
  class="nav"
>
  <ul
    class="topNav"
  >
    <li
      class="navItem firstElem"
    >
      <a
        href="/user/apps"
      >
        <img
          class="icon"
          src="/logo_icon.svg"
        />
      </a>
      <a
        href="/user/apps"
      >
        <label
          class="title"
        >
          QingCloud App Center
        </label>
      </a>
    </li>
  </ul>
  <ul
    class="bottomNav"
  >
    <li>
      <div
        class="tooltip popover iconOuter"
      >
        <div
          class="target"
        >
          <span
            class="icon icon-human icon"
            style="width:20px;height:20px"
          >
            <svg
              class="qicon qicon-human qicon-dark"
            >
              <use
                href="#qui-human"
              />
            </svg>
          </span>
          <label
            class="title"
          >
            My account
          </label>
        </div>
        <div
          class="popper pi-popover-popper"
          style="position:absolute;pointer-events:none;opacity:0"
        >
          <div>
            <div
              class="pi-popover-content"
            >
              <ul
                class="menuLayer"
              >
                <li>
                  <span
                    class="userIcon"
                  >
                    <span
                      class="icon icon-human iconImg"
                      style="width:20px;height:20px"
                    >
                      <svg
                        class="qicon qicon-human qicon-dark"
                      >
                        <use
                          href="#qui-human"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="username"
                  />
                  <span
                    class="devIconOuter undefined"
                  >
                    <span
                      class="icon icon-enterprise devIcon"
                      style="width:8px;height:8px"
                    >
                      <svg
                        class="qicon qicon-enterprise qicon-white"
                      >
                        <use
                          href="#qui-enterprise"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
                <li
                  class="dev"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-appcenter icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-appcenter qicon-dark"
                      >
                        <use
                          href="#qui-appcenter"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      App Center
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/account"
                  >
                    <span
                      class="icon icon-folder icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-folder qicon-dark"
                      >
                        <use
                          href="#qui-folder"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Account Info
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/password"
                  >
                    <span
                      class="icon icon-lock icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-lock qicon-dark"
                      >
                        <use
                          href="#qui-lock"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Change Password
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-loudspeaker icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-loudspeaker qicon-dark"
                      >
                        <use
                          href="#qui-loudspeaker"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Notice settings
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-creditcard icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-creditcard qicon-dark"
                      >
                        <use
                          href="#qui-creditcard"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Payment
                    </span>
                  </a>
                </li>
                <li
                  class="divider"
                >
                  <a
                    href="/logout"
                  >
                    <span
                      class="icon icon-previous icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-previous qicon-dark"
                      >
                        <use
                          href="#qui-previous"
                        />
                      </svg>
                    </span>
                    <span>
                      Log out
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Layout/SideNav basic render ISV portal 1`] = `
<div
  class="nav"
>
  <ul
    class="topNav"
  >
    <li
      class="navItem firstElem"
    >
      <a
        href="/user/apps"
      >
        <img
          class="icon"
          src="/logo_icon.svg"
        />
      </a>
      <a
        href="/user/apps"
      >
        <label
          class="title"
        >
          Provider Center
        </label>
      </a>
    </li>
    <li
      class="navItem"
    >
      <a
        href="/isv/apps"
      >
        <span
          class="icon icon-appcenter icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-appcenter qicon-dark"
          >
            <use
              href="#qui-appcenter"
            />
          </svg>
        </span>
      </a>
      <a
        href="/isv/apps"
      >
        <label
          class="title"
        >
          App Manage
        </label>
      </a>
    </li>
    <li
      class="navItem"
    >
      <a
        href="/dev/apps"
      >
        <span
          class="icon icon-wrench icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-wrench qicon-dark"
          >
            <use
              href="#qui-wrench"
            />
          </svg>
        </span>
      </a>
      <a
        href="/dev/apps"
      >
        <label
          class="title"
        >
          App Develop
        </label>
      </a>
    </li>
    <li
      class="navItem disabled"
    >
      <a
        href="/"
      >
        <span
          class="icon icon-ticket icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-ticket qicon-dark"
          >
            <use
              href="#qui-ticket"
            />
          </svg>
        </span>
      </a>
      <a
        href="/"
      >
        <label
          class="title"
        >
          Work list
        </label>
      </a>
    </li>
    <li
      class="navItem disabled"
    >
      <a
        href="/"
      >
        <span
          class="icon icon-linechart icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-linechart qicon-dark"
          >
            <use
              href="#qui-linechart"
            />
          </svg>
        </span>
      </a>
      <a
        href="/"
      >
        <label
          class="title"
        >
          Operation Center
        </label>
      </a>
    </li>
    <li
      class="navItem disabled"
    >
      <a
        href="/"
      >
        <span
          class="icon icon-wallet icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-wallet qicon-dark"
          >
            <use
              href="#qui-wallet"
            />
          </svg>
        </span>
      </a>
      <a
        href="/"
      >
        <label
          class="title"
        >
          Financial Center
        </label>
      </a>
    </li>
    <li
      class="navItem"
    >
      <a
        href="/isv/provider"
      >
        <span
          class="icon icon-shield icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon-shield qicon-dark"
          >
            <use
              href="#qui-shield"
            />
          </svg>
        </span>
      </a>
      <a
        href="/isv/provider"
      >
        <label
          class="title"
        >
          Service Provider Detail
        </label>
      </a>
    </li>
  </ul>
  <ul
    class="bottomNav"
  >
    <li>
      <div
        class="tooltip popover iconOuter"
      >
        <div
          class="target"
        >
          <span
            class="icon icon-human icon"
            style="width:20px;height:20px"
          >
            <svg
              class="qicon qicon-human qicon-dark"
            >
              <use
                href="#qui-human"
              />
            </svg>
          </span>
          <label
            class="title"
          >
            My account
          </label>
        </div>
        <div
          class="popper pi-popover-popper"
          style="position:absolute;pointer-events:none;opacity:0"
        >
          <div>
            <div
              class="pi-popover-content"
            >
              <ul
                class="menuLayer"
              >
                <li>
                  <span
                    class="userIcon"
                  >
                    <span
                      class="icon icon-human iconImg"
                      style="width:20px;height:20px"
                    >
                      <svg
                        class="qicon qicon-human qicon-dark"
                      >
                        <use
                          href="#qui-human"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    class="username"
                  />
                  <span
                    class="devIconOuter undefined"
                  >
                    <span
                      class="icon icon-shield devIcon"
                      style="width:8px;height:8px"
                    >
                      <svg
                        class="qicon qicon-shield qicon-white"
                      >
                        <use
                          href="#qui-shield"
                        />
                      </svg>
                    </span>
                  </span>
                </li>
                <li
                  class="dev"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-appcenter icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-appcenter qicon-dark"
                      >
                        <use
                          href="#qui-appcenter"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      App Center
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/account"
                  >
                    <span
                      class="icon icon-folder icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-folder qicon-dark"
                      >
                        <use
                          href="#qui-folder"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Account Info
                    </span>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    href="/user/profile/password"
                  >
                    <span
                      class="icon icon-lock icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-lock qicon-dark"
                      >
                        <use
                          href="#qui-lock"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Change Password
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-loudspeaker icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-loudspeaker qicon-dark"
                      >
                        <use
                          href="#qui-loudspeaker"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Notice settings
                    </span>
                  </a>
                </li>
                <li
                  class="disabled"
                >
                  <a
                    href="/"
                  >
                    <span
                      class="icon icon-creditcard icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-creditcard qicon-dark"
                      >
                        <use
                          href="#qui-creditcard"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Payment
                    </span>
                  </a>
                </li>
                <li
                  class="divider"
                >
                  <a
                    href="/user/runtimes"
                  >
                    <span
                      class="icon icon-image icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-image qicon-dark"
                      >
                        <use
                          href="#qui-image"
                        />
                      </svg>
                    </span>
                    <span
                      class="label"
                    >
                      Testing env
                    </span>
                  </a>
                </li>
                <li
                  class="divider"
                >
                  <a
                    href="/logout"
                  >
                    <span
                      class="icon icon-previous icon"
                      style="width:16px;height:16px"
                    >
                      <svg
                        class="qicon qicon-previous qicon-dark"
                      >
                        <use
                          href="#qui-previous"
                        />
                      </svg>
                    </span>
                    <span>
                      Log out
                    </span>
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`Layout/SideNav developer portal nav 1`] = `
Array [
  <div
    class="nav"
  >
    <ul
      class="topNav"
    >
      <li
        class="navItem firstElem"
      >
        <a
          href="/user/apps"
        >
          <img
            class="icon"
            src="/logo_icon.svg"
          />
        </a>
        <a
          href="/user/apps"
        >
          <label
            class="title"
          >
            QingCloud App Center
          </label>
        </a>
      </li>
      <li
        class="navItem"
      >
        <a
          class="iconLink"
          href="/user/apps/app_id/versions"
        >
          <img
            class="img image"
            data-origin-url="icon"
            src="icon"
          />
        </a>
        <a
          href="/user/apps/app_id/versions"
        >
          <label
            class="title"
          >
            name
          </label>
        </a>
      </li>
      <li
        class="navItem"
      >
        <a
          href="/user/apps/create"
        >
          <span
            class="icon icon-add icon"
            style="width:20px;height:20px"
          >
            <svg
              class="qicon qicon-add qicon-dark"
            >
              <use
                href="#qui-add"
              />
            </svg>
          </span>
        </a>
        <a
          href="/user/apps/create"
        >
          <label
            class="title"
          >
            Create app
          </label>
        </a>
      </li>
      <li
        class="navItem"
      >
        <a
          href="/user/apps"
        >
          <span
            class="icon icon-more icon"
            style="width:20px;height:20px"
          >
            <svg
              class="qicon qicon-more qicon-dark"
            >
              <use
                href="#qui-more"
              />
            </svg>
          </span>
        </a>
        <a
          href="/user/apps"
        >
          <label
            class="title"
          >
            View all
          </label>
        </a>
      </li>
    </ul>
    <ul
      class="bottomNav userBottomNav"
    >
      <li>
        <div
          class="tooltip popover iconOuter"
        >
          <div
            class="target"
          >
            <span
              class="icon icon-human icon"
              style="width:20px;height:20px"
            >
              <svg
                class="qicon qicon-human qicon-dark"
              >
                <use
                  href="#qui-human"
                />
              </svg>
            </span>
            <label
              class="title"
            >
              My account
            </label>
          </div>
          <div
            class="popper pi-popover-popper"
            style="position:absolute;pointer-events:none;opacity:0"
          >
            <div>
              <div
                class="pi-popover-content"
              >
                <ul
                  class="menuLayer"
                >
                  <li>
                    <span
                      class="userIcon"
                    >
                      <span
                        class="icon icon-human iconImg"
                        style="width:20px;height:20px"
                      >
                        <svg
                          class="qicon qicon-human qicon-dark"
                        >
                          <use
                            href="#qui-human"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      class="username"
                    />
                    <span
                      class="devIconOuter undefined"
                    >
                      <span
                        class="icon icon-wrench devIcon"
                        style="width:8px;height:8px"
                      >
                        <svg
                          class="qicon qicon-wrench qicon-white"
                        >
                          <use
                            href="#qui-wrench"
                          />
                        </svg>
                      </span>
                    </span>
                  </li>
                  <li
                    class="dev"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="icon icon-appcenter icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-appcenter qicon-dark"
                        >
                          <use
                            href="#qui-appcenter"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        App Center
                      </span>
                    </a>
                  </li>
                  <li
                    class=""
                  >
                    <a
                      href="/user/profile/account"
                    >
                      <span
                        class="icon icon-folder icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-folder qicon-dark"
                        >
                          <use
                            href="#qui-folder"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Account Info
                      </span>
                    </a>
                  </li>
                  <li
                    class=""
                  >
                    <a
                      href="/user/profile/password"
                    >
                      <span
                        class="icon icon-lock icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-lock qicon-dark"
                        >
                          <use
                            href="#qui-lock"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Change Password
                      </span>
                    </a>
                  </li>
                  <li
                    class="disabled"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="icon icon-loudspeaker icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-loudspeaker qicon-dark"
                        >
                          <use
                            href="#qui-loudspeaker"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Notice settings
                      </span>
                    </a>
                  </li>
                  <li
                    class="disabled"
                  >
                    <a
                      href="/"
                    >
                      <span
                        class="icon icon-creditcard icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-creditcard qicon-dark"
                        >
                          <use
                            href="#qui-creditcard"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Payment
                      </span>
                    </a>
                  </li>
                  <li
                    class="divider"
                  >
                    <a
                      href="/user/runtimes"
                    >
                      <span
                        class="icon icon-image icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-image qicon-dark"
                        >
                          <use
                            href="#qui-image"
                          />
                        </svg>
                      </span>
                      <span
                        class="label"
                      >
                        Testing env
                      </span>
                    </a>
                  </li>
                  <li
                    class="divider"
                  >
                    <a
                      href="/logout"
                    >
                      <span
                        class="icon icon-previous icon"
                        style="width:16px;height:16px"
                      >
                        <svg
                          class="qicon qicon-previous qicon-dark"
                        >
                          <use
                            href="#qui-previous"
                          />
                        </svg>
                      </span>
                      <span>
                        Log out
                      </span>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  <div
    class="subNav"
  >
    <div
      class="title"
    >
      Dashboard
    </div>
    <a
      class="link"
      href="/"
    >
      Overview
    </a>
  </div>,
]
`;
